internal/ips/retrieve.go

```diff
@@ -101,11 +100,12 @@ func (c *Client) Retrieve() *cobra.Command {
 				return c.Out.Output(ip, header, &data)
 			}
 
-			ips, err := pager.GetAllIPReservations(c.IPService, projectID, inc, exc, types)
+			// Don't have a paginator for this one
+			resp, _, err := c.IPService.FindIPReservations(context.Background(), projectID).Include(inc).Exclude(exc).Types(types).Execute()
```

Comment:
Seems like we need to patch in a paginator for the IP reservations endpoint (and maybe others?).  IIRC we had a python script in metal-go for identifying paginatable endpoints so maybe we can set up equinix-sdk-go to automatically find and patch these things.

Reply:
Actually, on closer inspection the old CLI-specific paginator was also not paginating; if it ever got a response that had more than one page it would just duplicate the first page over and over: https://github.com/equinix/metal-cli/blob/671da79d4fab26d68b0f430cd0ad12e35bb738f6/internal/pagination/pager.go#L155

The page parameter isn't included in the spec for the find IP reservations endpoint but I don't know if that's a spec bug or if it's intentional; @sbhatnagar-equinix do you know if the IP reservations endpoint is supposed to support pagination?

Reply:
Related issue in equinix-sdk-go: equinix/equinix-sdk-go#18

Reply:
Worked around this issue for now by upgrading to 0.31.2, but the upstream defect still exists.